### DISCUSSION TITLE
Identical IP addresses for clients/agents on Meshcentral deployed in Docker.

### DISCUSSION THREAD

**Discussion by Creogenesis** [Ylianst/MeshCentral#5763] [Q&A] [answered]
Everything works fine except for the display of IP addresses of the machines on which the agents are running. If I understand correctly, their real addresses should be displayed and they should be different. I think the problem might be with some firewall permissions on my server. 

![Screenshot_20240202_100607](https://github.com/Ylianst/MeshCentral/assets/13987755/8310a0be-4ddc-44a0-a122-66f7b7720916)

Please help with this issue. Perhaps someone who encounters this can use this discussion as a solution with a ready-made answer.

Additionally, I note that the IntelAMT status on clients began to be displayed after opening the IN OUT port 4433.

Host system: **Almalinux 8**
Web server: **Apache+Nginx** (nginx proxy to docker port meshcentral)
Latest version of Docker (docker swarm - single node)
Latest version of Portainer

**Comment by si458** [comment]
you need to set `trustedProxy` inside `settings` to the IP address of your reverse proxy
otherwise meshcentral will only show the IP address of your reverseproxy
![image](https://github.com/Ylianst/MeshCentral/assets/765314/d4af4d75-e949-4dfa-996b-0b5e698fb140)

**Reply by Creogenesis** [reply]
How is this?

trustedProxy: "XXX.XX.XX.XXX",
other option
}
},

Where XXX.XX.XX.XXX is my external IP address of the server where nginx is, or is it some other address?

I'm not very good at English, sorry.

**Reply by si458** [reply, marked as answer]
yes you put `trustedProxy` inside of `settings` in your `config.json`
and yes put the IP address of your reverse proxy/nginx container/server! NOT THE EXTERNAL IP! 192.168 10.0. 172.16 etc...

EXAMPLE:
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "myserver.mydomain.com",
    "trustedProxy": "192.168.0.99",
    "_WANonly": true,
    "_LANonly": true,
    "sessionKey": "MyReallySecretPassword1",
    "port": 443,
    "redirPort": 80,
  },
  "domains": {
    "": {
      "title": "MyServer",
      "newAccounts": false
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```

**Reply by Creogenesis** [reply]
This is my schema 

NGINX proxy pass
![Screenshot_20240202_142903](https://github.com/Ylianst/MeshCentral/assets/13987755/acfafdd5-358c-42eb-9129-c1991bfb49ca)

Docker networks
![Screenshot_20240202_142620](https://github.com/Ylianst/MeshCentral/assets/13987755/84b1df3a-bd29-4088-b3bb-107af8edc57a)


And IP can maybe changed after reboot
![Screenshot_20240202_142748](https://github.com/Ylianst/MeshCentral/assets/13987755/28cabb36-3faa-4a31-b413-bec43a67b71a)

**My nginx is not deployed in Docker; it is an external host web server that also serves other purposes.**
**What then in my case?**

**Reply by si458** [reply]
then just use the IP address that is shown in the meshcentral as that must be the IP address of your nginx host `10.0.0.2`

**Reply by Creogenesis** [reply]
I will try now.

**Reply by Creogenesis** [reply]
**Simply wonderful! Everything works.** Your method of specifying the IP that is displayed in the dashboard in the client list has become a kind of lifehack for those with my method of deployment in Docker.
